statina/API/v2/endpoints/batches.py

```diff
@@ -0,0 +1,288 @@
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Literal, Optional
+
+from fastapi import APIRouter, Depends, Form, Query, Security, Response
+from fastapi.encoders import jsonable_encoder
+from fastapi.responses import FileResponse, JSONResponse, StreamingResponse
+
+import statina.crud.find.plots.fetal_fraction_plot_data as get_fetal_fraction
+from statina.adapter import StatinaAdapter
+from statina.API.external.constants import TRISOMI_TRESHOLDS
+from statina.API.v2.endpoints.user import get_current_active_user
+from statina.config import get_nipt_adapter
+from statina.crud import update
+from statina.crud.delete import delete_batch
+from statina.crud.find import find
+from statina.crud.find.plots.coverage_plot_data import (
+    get_box_data_for_coverage_plot,
+    get_scatter_data_for_coverage_plot,
+)
+from statina.crud.find.plots.zscore_plot_data import (
+    get_tris_control_abnormal,
+    get_tris_control_normal,
+    get_tris_samples,
+)
+from statina.crud.insert import insert_batch, insert_samples
+from statina.crud.utils import zip_dir
+from statina.models.database import Batch, DataBaseSample, User
+from statina.models.server.load import BatchRequestBody
+from statina.models.server.plots.coverage import CoveragePlotSampleData
+from statina.models.server.plots.fetal_fraction import (
+    FetalFractionControlAbNormal,
+    FetalFractionSamples,
+)
+from statina.models.server.plots.fetal_fraction_sex import SexChromosomeThresholds
+from statina.models.server.sample import Sample
+from statina.parse.batch import get_batch, get_samples, validate_file_path
+
+router = APIRouter(prefix="/v2")
+
+
+@router.get("/batches")
+def batches(
+    page_size: Optional[int] = Query(5),
```

Comment:
Quess we will have to add the search queries to combine pagination with user interactive search